webrender/src/display_list_flattener.rs

```diff
@@ -638,12 +639,29 @@ impl<'a> DisplayListFlattener<'a> {
                     Some(tiling) => {
                         // The image resource is tiled. We have to generate an image primitive
                         // for each tile.
-                        self.decompose_image(
-                            clip_and_scroll,
-                            &prim_info,
-                            info,
-                            tiling.image_size,
-                            tiling.tile_size as u32,
+                        decompose_image(
+                            &TiledImageInfo {
+                                rect: prim_info.rect,
+                                tile_spacing: info.tile_spacing,
+                                stretch_size: info.stretch_size,
+                                device_image_size: tiling.image_size,
+                                device_tile_size: tiling.tile_size as u32,
+                            },
+                            &mut|tile| {
```

Comment:
Instead of passing a callback here, would it be possible to make the new function part of a DisplayListFlattener impl in the `image.rs` file? This is how `box_shadow.rs` and `border.rs` are structured.

Reply:
The goal is to decouple this code entirely from `DisplayListFlattener` so that in the next step I move it to the frame building phase (and make it operate only on the visible area).